sycl/source/detail/program_impl.cpp

```diff
@@ -158,7 +157,7 @@ program_impl::program_impl(ContextImplPtr Context,
 }
 
 program_impl::program_impl(ContextImplPtr Context, RT::PiKernel Kernel)
-    : program_impl(Context, nullptr,
+    : program_impl(Context, pi_native_handle{},
```

Comment:
can  you be explicit and use null to initialize it?

Reply:
I've inserted an explicit cast of nullptr. A bit more verbose, but I agree it's the better option.